### PR TITLE
feat(vats): allow replacing a scoped bridge handler

### LINKED ISSUE #7576: `BridgeScopedManager` should allow transferring to new handler
## What is the Problem Being Solved?
In order to sidestep upgradability of the bootstrap vat (#4548), we [sketched a plan](https://github.com/Agoric/agoric-sdk/blob/15fcdbce433db30f3439554e06e6161b830e4ab5/packages/vats/src/core/README.md#upgrade) to transition bootstrap powers to a new vat, and manually rewire all exported objects. One of those cases is the core eval bridge handler, registered with the BridgeManager (defined in the bridge / chain storage vat).

## Description of the Design

Currently the way to do register a handler is through a `setHandler` method on the `BridgeScopedManager`, which is protected by a one time write check. I forgot the reasoning behind the check, but it was likely to avoid mistaken re-registration. We should either relax this check (maybe by forcing a trip through a de-registration followed by a re-registration), or change the way handlers are registered (there is an argument for having 2 facets to the scoped manager).

## Security Considerations

Setting a new handler is no different than replacing the handler's implementation to forward to another handler, which you ought to be allowed to do as the handler provider.

## Scaling Considerations

None

## Test Plan

TBD

### PR DESCRIPTION
**Pull request by mhofman**
closes: #7576

## Description

Supports future upgrade paths where the handler was not durable by allowing the handler to be replaced, but only in cases where the old handler can be provided to prevent unexpected changes.

### Security Considerations

Requiring the old handler effectively restricts the capability to change handlers to the previous provider, which could have done that in the first place by updating its implementation. No new power is granted through this change.

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

The bridge manager does not have unit tests and I think it wouldn't be cost effective to add them as part of this simple change.

### CODE REVIEW

**Review by dckc — APPROVED** [review]
yes, let's

File: packages/vats/src/bridge.js [hunk 1]
```diff
@@ -61,12 +63,14 @@ const prepareScopedManager = zone => {
...
         // setHandler could probably be on a separate facet to separate it
         // from the toBridge and fromBridge powers, but it's easier to
-        // implement a set once check for the handler and pass a single
+        // implement an already set check for the handler and pass a single
         // object around.
-        !this.state.inboundHandler ||
+        // We do allow explicitly replacing the handler by requiring the old
+        // handler to also be provided.
+        this.state.inboundHandler === oldHandler ||
           Fail`Bridge handler already set for ${this.state.bridgeId}`;
```
Comment (gibson042) [inline packages/vats/src/bridge.js:74]: This error message will be misleading when attempting a failed replacement, but only slightly—I think keeping it is probably fine.

**Review by gibson042 — APPROVED** [review]

File: packages/vats/src/bridge.js [hunk 2]
```diff
@@ -61,12 +63,14 @@ const prepareScopedManager = zone => {
         assert(inboundHandler, X`No inbound handler for ${bridgeId}`);
         return E(inboundHandler).fromBridge(obj);
       },
-      setHandler(newHandler) {
+      setHandler(newHandler, oldHandler) {
```
Comment (michaelfig) [inline packages/vats/src/bridge.js:66]: We should not require an `oldHandler` argument, as discussed internally.  Can you please revert this addition?

File: packages/vats/src/bridge.js (hunk 1 above)
Comment (michaelfig) [reply packages/vats/src/bridge.js:74]: Please remove this assertion: it doesn't provide any clear benefit, and just makes the API for replacement more clumsy.

**Pull request merged**